### PR TITLE
[DI2-59] Replace Redux router links with React router

### PR DESCRIPTION
**Pull request by jonkoops**

### CODE REVIEW

**Review by chrisvanmook — APPROVED** [review]

File: src/app/links.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import { LocationDescriptor } from 'history'
+import { generatePath } from 'react-router-dom'
+import { ViewMode } from '../shared/ducks/ui/ui'
+import parameters from '../store/parameters'
+import { routing } from './routes'
+
+export const toConstructionFile = (
```
Comment (janjaap) [inline src/app/links.ts:7]: The name of the functions in this module are confusing; they suggest performing an action, but instead return an object that can be object as the value of the `to` prop on a `Link` component. Maybe `getConstructionFileLocation`?

File: src/app/links.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { generatePath } from 'react-router-dom'
+import { toAddresses, toCadastralObjects, toConstructionFile, toEstablishments } from './links'
+import { routing } from './routes'
+
+describe('toConstructionFile', () => {
+  it('creates a link', () => {
```
Comment (janjaap) [inline src/app/links.test.ts:6]: Doesn't create a link, but return location properties that can be used to construct a link from.

**Review by janjaap — CHANGES_REQUESTED** [review]
There's about twenty other components that make use of `redux-first-router-link`. Will those be tackled in this PR as well?

**Comment by jonkoops** [comment]
> There's about twenty other components that make use of `redux-first-router-link`. Will those be tackled in this PR as well?

This was originally the plan, but we decided to split this up into multiple separate tasks so those will come as separate PRs.

File: src/app/pages/MapPage/draw/DrawResults.tsx [hunk 3]
```diff
@@ -99,7 +99,7 @@ const StyledLabel = styled(Label)`
   }
 `
 
-const TableRouterLink = styled(RouterLink)`
+const TableRouterLink = styled(ReactRouterLink)`
```
[the remark this answers is not in the document]
Comment (jonkoops) [reply src/app/pages/MapPage/draw/DrawResults.tsx:102]: That import cannot be removed yet as it is still used for some code (see line 45-48).

```ts
const ResultLink = styled(RouterLink)`
  width: 100%;
  margin-bottom: ${themeSpacing(2)};
`
```

File: src/app/links.ts (hunk 1 above)
Comment (jonkoops) [reply src/app/links.ts:7]: I agree that the naming is not fantastic but I think it makes sense in the context it's used. I'd like to keep it like this for now, we can always rename them all at the end.

File: src/app/links.test.ts (hunk 2 above)
Comment (jonkoops) [reply src/app/links.test.ts]: Changed this to 'location descriptor' like the type it returns.

**Pull request merged**